### ISSUE TITLE
v1.2.7 breaks action for GHES 3.7.7

### ISSUE THREAD

**Issue by jedla01** [actions/deploy-pages#137]
We use GitHub Enterprise Server 3.7.7 in our org and the v1.2.7 update breaks workflow for us. Had to specify v1.2.6 to use.

With 1.2.7, the action fails with:
```
Error: Creating Pages deployment failed
Error: HttpError: Not Found
    at /runner/_work/_actions/actions/deploy-pages/v1/webpack:/deploy-pages/node_modules/@octokit/request/dist-node/index.js:86:1
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at createPagesDeployment (/runner/_work/_actions/actions/deploy-pages/v1/webpack:/deploy-pages/src/api-client.js:116:1)
    at Deployment.create (/runner/_work/_actions/actions/deploy-pages/v1/webpack:/deploy-pages/src/deployment.js:58:1)
    at main (/runner/_work/_actions/actions/deploy-pages/v1/webpack:/deploy-pages/src/index.js:30:1)
Error: Error: Failed to create deployment (status: 404) with build version b016166ec21b31bee21e3bff29219a2dd107[18](https://xxx/org/repo/runs/3672925?check_suite_focus=true#step:3:19)5c. Ensure GitHub Pages has been enabled: https://xxx/org/repo/runs/settings/pages
```
I've deducted the url details.

Understand that the fix is for the main Github cloud. Do you plan any backwards compatibility for the previous versions of GHES?

**Comment by yoannchaudet** [comment]
@jedla01 Sorry for that, I don't think that was our intention to break GHES.

@JamesMGreene @TooManyBees Could it be related to the change on the non-documented API?

**Comment by JamesMGreene** [comment]
Yeah, looks like that would do it. 😓 

Suggested remediation:
- [ ] Publish a new `v1.x` to "revert" that for folks referencing `@v1`
- [ ] Re-release the change in `v1.2.7` as `v2.0.0` to make these distinctions cleaner

**Comment by jedla01** [comment]
> @jedla01 Thanks for this speedy report! We've published a [`v1.2.8` release](https://github.com/actions/deploy-pages/releases/v1.2.8, as well as updating the `v1` tag, to revert the breaking change in `v1.2.7`. 🙇🏻
> 
> Please let us know if that doesn't resolve your issue for some reason. 👍🏻

Thank you for such fast response and fix.

**Comment by Web4application** [comment]
@dependabot
